src/dungeongen.cpp

```diff
@@ -107,23 +111,23 @@ void DungeonGen::generate(MMVManip *vm, u32 bseed, v3s16 nmin, v3s16 nmax)
 		}
 	}
 
-	// Add it
-	makeDungeon(v3s16(1, 1, 1) * MAP_BLOCKSIZE);
+	// Add them
+	for (u32 i = 0; i < floor(nval_density); i++)
+		makeDungeon(v3s16(1, 1, 1) * MAP_BLOCKSIZE);
 
-	// Convert some cobble to mossy cobble
-	if (dp.mossratio != 0.0) {
-		for (s16 z = nmin.Z; z <= nmax.Z; z++)
-		for (s16 y = nmin.Y; y <= nmax.Y; y++) {
-			u32 i = vm->m_area.index(nmin.X, y, z);
-			for (s16 x = nmin.X; x <= nmax.X; x++) {
-				if (vm->m_data[i].getContent() == dp.c_cobble) {
-					float wetness = NoisePerlin3D(&dp.np_wetness, x, y, z, dp.seed);
-					float density = NoisePerlin3D(&dp.np_density, x, y, z, blockseed);
-					if (density < wetness / dp.mossratio)
-						vm->m_data[i].setContent(dp.c_moss);
-				}
-				i++;
+	// Optionally convert some structure to alternative structure
+	if (dp.c_alt_wall == 0)
```

Comment:
Previously 'mossratio' was used to choose whether the alt_wall-placng loop was run. Now we need a new way to prevent the alt_wall-placing loop from running when no alt_wall node is wanted.